File: frontend/src/services/hospitalService.js
import apiClient from './apiClient';

export const hospitalService = {
  async getAllHospitals(search = '') {
    const params = search ? { search } : {};
    const response = await apiClient.get('/hospitals', { params });
    return response.data;
  },

  async getHospitalById(id) {
    const response = await apiClient.get(`/hospitals/${id}`);
    return response.data;
  },

  async getDepartments(hospitalId) {
    const response = await apiClient.get(`/hospitals/${hospitalId}/departments`);
    return response.data;
  },
};

File: backend/src/main/java/com/yourorg/appname/service/HospitalService.java
package com.yourorg.appname.service;

import com.yourorg.appname.dto.response.DepartmentResponse;
import com.yourorg.appname.dto.response.HospitalResponse;

import java.util.List;

public interface HospitalService {
    List<HospitalResponse> getAllHospitals(String search);
    HospitalResponse getHospitalById(Long id);
    List<DepartmentResponse> getDepartmentsByHospital(Long hospitalId);
}
